### DISCUSSION TITLE
ps4 controller + esp32

### DISCUSSION THREAD

**Discussion by freevedo** [micropython/micropython#11287] [ESP32]
Hello everyone. I'm looking for a solution to connect my esp32 with a ps4 controller using micropython. There is out there a lot of arduino code to do that.
My project is for the university program. I am almost done(I used micropython from all the project) and going back to arduino is not an option.

**Comment by karfas** [comment]
> I'm looking for a solution to connect my esp32 with a ps4 controller

And your problem is ... ?

Just look what the arduino code is doing and try to do similar things in python.

**Reply by freevedo** [reply]
I will need some time to rewrite it. It is a little bit urgent. I wanted to build my own controller but due to a lack of time I decided to use the ps4 controller(I had already work with it).

**Comment by freevedo** [comment]
On one hand the PS4 controller does not support Bluetooth Low Energy, on the other Hand micropython only supports Bluetooth low Energy. It is a not compatible.

**Reply by Diegolox** [reply]
Bro, it's not correct. Im using a PS4 controller in my ESP32 with out any problem. But with out BLE.

**Comment by davidgbustamante** [comment]
Do you mean that it's not possible to do it through MicroPython?

**Reply by Diegolox** [reply]
It's posible, but u have to do it.
